src/controller.cc

```diff
@@ -211,6 +215,32 @@ void Controller::ScheduleTransaction() {
         auto cmd = TransToCommand(*it);
         if (cmd_queue_.WillAcceptCommand(cmd.Rank(), cmd.Bankgroup(),
                                          cmd.Bank())) {
+
+            // update stats for all pending requests
+            if (cmd.IsWrite()) {
+                auto num_requests = pending_wr_q_.count(cmd.hex_addr);
```

Comment:
Why are you doing this for all requests in the queue?

Reply:
This is not necessary for writes as there will be one write from "write_buffer_" for a given address in "pending_wr_q_ ". In other words, there will be one transaction "tran" in the "write_buffer_" with address "addr" and there will one entry in "pending_wr_q_" with address "addr" mapped to the same transaction "trans".
However, for reads one entry in "read_queue_" can represent multiple transactions in "pending_rd_q_". For example, we can have <addr, tran0> and <addr, tran1> in "pending_rd_q_" with the same address and just one entry "tran0" in "read_queue_". Thus, upon scheduling, we need to update the "schedule_cycle" for all of the transaction in "pending_rd_q_" to make sure the **stats we are collecting per transaction** would be correct.